test/e2e/storage/persistent_volumes-local.go

```diff
@@ -53,96 +54,238 @@ const (
 	// TODO: This may not be available/writable on all images.
 	hostBase      = "/tmp"
 	containerBase = "/myvol"
-	testFile      = "test-file"
-	testContents  = "testdata"
-	testSC        = "local-test-storagclass"
+	// Path to the first volume in the test containers
+	// created via createLocalPod or makeLocalPod
+	// leveraging pv_util.MakePod
+	volumeDir = "/mnt/volume1"
+	// testFile created in setupLocalVolume
+	testFile = "test-file"
+	// testFileContent writtent into testFile
+	testFileContent = "test-file-content"
+	testSC          = "local-test-storageclass"
 )
 
-var _ = framework.KubeDescribe("[Volume] PersistentVolumes-local [Feature:LocalPersistentVolumes] [Serial]", func() {
+var _ = framework.KubeDescribe("PersistentVolumes-local [Volume] [Feature:LocalPersistentVolumes] [Serial]", func() {
 	f := framework.NewDefaultFramework("persistent-local-volumes-test")
 
 	var (
 		config *localTestConfig
+		node0  *v1.Node
 	)
 
 	BeforeEach(func() {
+
+		// Get all the schedulable nodes
+		nodes := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		Expect(len(nodes.Items)).NotTo(BeZero(), "No available nodes for scheduling")
+
 		config = &localTestConfig{
 			ns:     f.Namespace.Name,
 			client: f.ClientSet,
-			nodes:  []v1.Node{},
-		}
-
-		// Get all the schedulable nodes
-		nodes, err := config.client.CoreV1().Nodes().List(metav1.ListOptions{})
-		if err != nil {
-			framework.Failf("Failed to get nodes: %v", err)
+			nodes:  nodes,
 		}
 
-		for _, node := range nodes.Items {
-			if !node.Spec.Unschedulable {
-				// TODO: does this need to be a deep copy
-				config.nodes = append(config.nodes, node)
-			}
-		}
-		if len(config.nodes) == 0 {
-			framework.Failf("No available nodes for scheduling")
-		}
+		// Choose the first node
+		node0 = &config.nodes.Items[0]
 	})
 
 	Context("when one pod requests one prebound PVC", func() {
 		var (
 			testVol *localTestVolume
-			node    *v1.Node
 		)
```

Comment:
very minor nit, while there were 2 variable declared here - old style made sense but with just one variable, it looks out of place